app/code/core/Mage/Dataflow/Model/Profile.php

```diff
@@ -159,7 +159,7 @@ protected function _afterSave()
             ->setActionCode($this->getOrigData('profile_id') ? 'update' : 'create')
             ->save();
         $csvParser = new Varien_File_Csv();
-        $delimiter = trim($this->getData('gui_data/parse/delimiter'));
+        $delimiter = trim((string)$this->getData('gui_data/parse/delimiter'));
```

Comment:
if delimiter is null, then becomes '', shouldn't be a different default value? or is the right default generated later?

Reply:
Sorry what is different about the logic than before? It's the same as PHP does internally (converting the `null` value to string)..

Reply:
I understand that but I was wondering if that could be fixed in a cleaner way (not cleaner than yours but cleaner than the original implementation)